language/move-prover/tests/sources/functional/global_vars.move

```diff
@@ -1,162 +1,145 @@
 module 0x42::TestGlobalVars {
 
-    /*
-    TODO(refactoring): this test is deactivated until we have ported this (or a similar) feature, or decided to
-      drop it in which case the test should be removed.
+    use Std::Signer;
 
-    spec module {
-        pragma verify = true;
-    }
+    // ================================================================================
+    // Counting
 
     spec module {
-        global sum_of_T: u64;
+        global sum_of_T: u64 = 0;
     }
 
-    resource struct T {
+    struct T has key {
       i: u64,
     }
-    spec T {
-      invariant pack sum_of_T = sum_of_T + i;
-      invariant unpack sum_of_T = sum_of_T - i;
+
+    fun add() acquires T {
+        borrow_global_mut<T>(@0).i = borrow_global_mut<T>(@0).i + 1
+    }
+    spec add {
+        update sum_of_T = sum_of_T + 1;
```

Comment:
Nice to see this feature! I have some questions. Is the expression that comes after `update` (e.g., `sum_of_T = sum_of_T + 1`) a relation or an assignment? Do we expect the update to happen at the end of the function? Perhaps, in some cases, it may be useful to allow controlling the update timing (e.g., `update [pre]`, `update [post]`)?

Reply:
It's an assignment, which is evaluated in the post state, but before any post conditions (including aborts_if) and invariants are evaluated.  I'm not sure whether we need the `update[pre/post]` thing; if possible, I like to avoid adding more complexity here.